### PR TITLE
use tolist instead of numpy for extracting parameters in onnx

### PR DESCRIPTION
**Pull request by geohotstan**

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -222,8 +222,6 @@ def tolist(self) -> Union[Sequence[ConstType], ConstType]:
     """
     Returns the value of this tensor as a nested list.
 
-    Currently this only works for flattened tensors.
-
```
Comment (geohotstan) [inline tinygrad/tensor.py:226]: does supporting nested lists mean tensors that aren't flat also work?
image

File: extra/onnx.py [hunk 2]
```diff
@@ -13,16 +14,16 @@
   from onnx.mapping import TENSOR_TYPE_TO_NP_TYPE
   tensor_dtype_to_np_dtype = lambda x: TENSOR_TYPE_TO_NP_TYPE[x]
 
-# global numpy cache for parameters
-numpy_cache = {}
-def safe_numpy(t) -> np.ndarray:
+# global Tensor -> list cache for parameters
+to_python_cache = {}
```
Comment (chenyuxyz) [inline extra/onnx.py:18]: can it work with `@functools.lru_cache` and don't need to manually maintain cache?

File: extra/onnx.py [hunk 3]
```diff
@@ -13,16 +14,16 @@
...
+def safe_python(t, tobytes=False) -> ConstType:
```
Comment (chenyuxyz) [inline extra/onnx.py:19]: return type does not seem correct, it should at least union with bytes? also can we rename the function to `to_python_const` or similar? it's not clear to me what's unsafe here

**Review by chenyuxyz — COMMENTED** [review]
very happy to see this cleanup!

**Pull request merged**